Redirect rapids.ai/cloud to the RAPIDS Cloud Deployment docs and other redirect fixes

jacobtomlinson opened the pull request:
With the 22.12 release the [RAPIDS Cloud Deployment documentation](https://docs.rapids.ai/deployment/stable/cloud/) now contains a superset of the information on rapids.ai/cloud and will be maintained and updated going forwards with each release.

It was raised during the developer review that the instructions on rapids.ai/cloud were out of date and the reviewers didn't find the new documentation at all, so this PR redirects the rapids.ai/cloud page over to the new docs.

Closes https://github.com/rapidsai/deployment/issues/67 and helps #265

ajschmidt8 commented:
@jacobtomlinson, just a heads up. We recently switched our hosting from GitHub Pages to [Netlify](https://www.netlify.com/).

Because of this, we can now commit a configuration file to our repository in order to configure server-side redirects (e.g. 301 redirects) instead of the client-side redirects that we did with GitHub Pages. Check out the Netlify docs below for more details!

- https://docs.netlify.com/routing/redirects/

jacobtomlinson commented:
Thanks @ajschmidt8 just circling back here.

I've updated the PR to use the Netlify redirect style. This also meant removing the existing `_redirects` collection due to the naming collision and updating the slack invite redirect to also use the Netlify style. I also saw the Sagemaker Studio Lab page had a redirect similar to the cloud one I proposed so I updated that too.

ajschmidt8 commented:
@exactlyallan beat me by a minute to update/merge this PR :upside_down_face:.

posted some additional changes here! https://github.com/rapidsai/rapids.ai/pull/287
